tests/util.py

```diff
@@ -397,141 +405,127 @@ def sort_inner_lists(obj):
-        settings.LORA_URL = 'http://localhost:{}/'.format(cls.lora_port())
-        settings.SAML_IDP_TYPE = None
-
-        # This is the first such measure: if the interpreter abruptly
-        # exits for some reason, tell the subprocess to exit as well
-        atexit.register(cls.minimox.send_signal, signal.SIGINT)
-
-        # wait for loading
-        s = requests.Session()
-        s.mount(
-            settings.LORA_URL,
-            requests.adapters.HTTPAdapter(
-                max_retries=urllib3.util.retry.Retry(
-                    10,
-                    backoff_factor=0.01,
-                ),
-            ),
-        )
+_psql_factory = testing.postgresql.PostgresqlFactory(
+    cache_initialized_db=True,
+    on_initialized=initdb
+)
+atexit.register(_psql_factory.clear_cache)
 
-        try:
-            s.get(settings.LORA_URL + 'site-map').raise_for_status()
-        except Exception as e:
-            raise unittest.SkipTest('LoRA startup failed!')
 
-    @classmethod
-    def tearDownClass(cls):
-        # first, we're cleaning up now, so clear the exit handler
-        atexit.unregister(cls.minimox.send_signal)
+class LoRATestCaseMixin(TestCaseMixin):
+    '''Base class for LoRA testcases; the test creates an empty LoRA
+    instance, and deletes all objects between runs.
+    '''
 
-        # second, terminate our child process
-        cls.minimox.send_signal(signal.SIGINT)
+    def load_sample_structures(self, **kwargs):
+        load_sample_structures(**kwargs)
 
-        settings.LORA_URL = cls._orig_lora
-        cls.minimox_log.close()
+    def lora_port(self):
+        return self.__lora_port
+
+    @classmethod
+    def get_lora_environ(cls):
+        '''Extra environment variables for the LoRA process.'''
+
+        return {}
 
     def setUp(self):
         super().setUp()
 
-        self.assertIsNone(self.minimox.poll(), 'LoRA suddenly died!')
+        self.psql = _psql_factory()
+        self.addCleanup(self.psql.stop)
 
-        self.flush_log()
+        self.psql.wait_booting()
 
-    def tearDown(self):
-        self.assertIsNone(self.minimox.poll(), 'LoRA suddenly died!')
+        dsn = self.psql.dsn()
 
-        self.flush_log()
+        from oio_rest import db
 
-        # delete all objects in the test instance; this does 'leak'
-        # information in that they continue to exist as registrations,
-        # but it's faster than recreating the database fully
-        c = lora.Connector(virkningfra='-infinity', virkningtil='infinity')
+        oio_rest.app.app.config["DEBUG"] = True
+        oio_rest.app.app.config["TESTING"] = True
+        oio_rest.app.app.config['PRESERVE_CONTEXT_ON_EXCEPTION'] = False
 
-        for t in map(c.__getattr__, c.scope_map):
-            for objid in t(bvn='%'):
-                t.delete(objid)
+        if hasattr(db.adapt, 'connection'):
+            del db.adapt.connection
+            self.addCleanup(lambda: db.adapt.connection.close())
```

Comment:
Shouldn't we close the connection regardless of whether it was opened before? With this, the cleanup function is not going to be registered on the first test run.

Reply:
> Shouldn't we close the connection regardless of whether it was opened before? With this, the cleanup function is not going to be registered on the first test run.

Erm, this change is overwritten by https://github.com/magenta-aps/mora/pull/104/commits/86cca32e1262fac1f3f587ec712bf9f03c25cd67?

Reply:
...I'm tired.